pkg/exchange/exchange.go

```diff
@@ -409,6 +481,36 @@ func (w *exchange) handleObjectRequests(subscription EnvelopeSubscription) error
 				peer.LookupByKey(e.Sender),
 				WithLocalDiscoveryOnly(),
 			)
+		case dataForwardType:
+			// if we receive a dataForward payload we check if it is meant for
+			// this peer, if yes try to decode it, if not forward it
+			// we treat this a data because it needs to be encrypted
+
+			// decode object
+			fwd := &DataForward{}
+			if err := fwd.FromObject((e.Payload)); err != nil {
+				return err
+			}
+
+			// todo: is this right?
```

Comment:
I think it’s ok for now as the data is not encrypted. But once they are we should be writing to the peer connection directly.